src/Aspire.Hosting.Azure/AzureCosmosDBCloudApplicationBuilderExtensions.cs

```diff
@@ -18,7 +18,7 @@ public static class AzureCosmosDBCloudApplicationBuilderExtensions
     /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
     /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
     /// <param name="connectionString">The connection string.</param>
-    /// <returns>A reference to the <see cref="IResourceBuilder{AzureCosmosDatabaseResource}"/>.</returns>
+    /// <returns>A reference to the <see cref="IResourceBuilder{AzureCosmosDBResource}"/>.</returns>
```

Comment:
As far as I know, XML docs don't allow generic specializations unfortunately. The type arguments here are the actual names of the parameters rather than the arguments in usage. So, this should just be:

```xml
    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
```

Ditto elsewhere here.

Reply:
I had to check and I can confirm. The generated xml files contains this:

```xml
<returns>A reference to the <see cref="T:Aspire.Hosting.ApplicationModel.IResourceBuilder`1"/>.</returns>
```

Whatever type we pass.

And then I checked IntelliSense, and the type doesn't matter either, using a valid class or `T` results in the same tooltip:

![image](https://github.com/dotnet/aspire/assets/1165805/9ec39860-8080-4e5e-9ebc-3572f45a3a6b)

But if the type in the docs doesn't exist we get a compilation error.

Now this change is replacing a class name with another class name. Technically, as you are stating, it doesn't change anything. But the one before was not even the returned type, so I still believe it's better. We still need to use `<see cref="IResourceBuilder{SOMETHING}"/>` so either use `T` or the actual returned type.

Or do you think using the actual type name from the generic definition (T) is better?

Reply:
I don't believe refactoring tools in VS will update these strings, if you rename the type.

Maybe someone reading the docs in the code would benefit from the instantiated type being given. But in this case it's just the return type which is given more naturally immediately below in the code, and in the tool tip.